exercises/sum-of-multiples/canonical-data.json

```diff
@@ -17,6 +17,13 @@
       "expected": 3
     },
     {
+      "description": "multiples of 3 or 5 up to 7",
+      "property": "sum",
+      "factors": [3, 5],
+      "limit": 7,
+      "expected": 14
```

Comment:
What is this testing that is different from the 3 or 5 up to 10 test?

It is good if the test describes what it is testing rather than repeating the input information.

Reply:
It was in my previous commit, but you had mentioned you liked the 2nd test case with the 5 as shown in the original, and I don't think it's a good idea to back and forth between inputting 1 factor and 2 factors. This test is supposed to be focusing on just summing the single factor.

Let me give a little overview how I see this problem getting solved, using the TPP.

Test #1 will simply be returning 0 (nil -> constant)
```code
    public static int Sum(int[] multiples, int max)
    {
        return 0;
    }
```

Test #2 will be splitting the execution path, adding the if statement for an input of 3 (statement -> conditional)
```code
    public static int Sum(IEnumerable<int> multiples, int max)
    {
        if(max > 3)
        {
            return 3;
        }

        return 0;
    }
```

Test #3 will be needing to Sum 3 and 6 to make 9. (statement ->conditional)
``` code
public static int Sum(IEnumerable<int> multiples, int max)
{
	if(max > 6)
	{
		return 9;
	}

	if(max > 3)
	{
		return 3;
	}

	return 0;
}
```
At this point, we have duplication. We have 3s and 6s, we're trying to add to the sum multiple times, we use if -> loop
```code
    public static int Sum(IEnumerable<int> multiples, int max)
    {
        int sum = 0;

        for (int x = 1; x < max; x++)
        {
            if (x % 3 == 0)
            {
                sum += x;
            }
        }

        return sum;
    }
```

At this point, we have solving for a factor of 3 implemented. The loop is there. The modulus is there. We just need to worry about introducing another factor and nothing more.

That's kind of where I'm trying to get with the introduction of a new case. I want to focus on the case where the only factor is 3 and the limit is 7. This case forces the student to sum 3 + 6 and not worry about an array of factors until Test #4. Before that, I like the idea of only having 3 as a factor.

Reply:
I have no problem with single factor cases. (I think they are a good idea.)
But they can be added to the start of the test cases without removing/changing any of the existing cases.

Reply:
(This was a pre-existing issue, so not your fault at all, but the test descriptions for this problem are not helpful.)

Reply:
Your TDD examples are helpful, but you are relying on the factor being 3 the whole time. Using a different approach I would build based on changing factors before changing limits, which leads to different tests being useful at different times.

What would you do if the the second and third tests were: 
` "factors": [3], "limit": 7,`
` "factors": [5], "limit": 7,`

Reply:
I think that would be a bit of a jump. We'd be going from returning 0, to adding 3 + 6. We'd be skipping the step where you include 3. I'd think you'd want to turn 0 into 3, then 3 + 6,

Reply:
OK, what about:
```
1. "factors": [3], "limit": 1,
2. "factors": [3], "limit": 3,
3. "factors": [3], "limit": 7,
4. "factors": [5], "limit": 7,
```

Reply:
Limit 4 would make more sense for the 2nd test, so we'll assume typo :)

But this was my initial proposal. So my tests are above. In order to make test 4 pass I would just change 3 to be more generic, multiples[0] (constant -> scalar)

Reply:
Right, so ADD these tests to the start of the json file.
But don't change/remove the existing tests, then everything should be fine.

Reply:
I feel like if we just added those four cases we'd just be bloating the test suite. If you pass those tests, the logic that we're verifying will have already been implemented. That's why I initially just removed 5 from the factors array and then introduced it in a later case.

Reply:
So what do you want to do with this PR?

I'm OK with a PR that only adds tests.
I'm OK with discussing a PR that removes tests.
I seem to be against a PR that does both.

What about turning this one into one that only adds new "simpler" tests, and then creating a new PR to remove the tests you then think are redundant?

Reply:
I'm not in a rush and I welcome the critique/discussions! I hopefully recruited a couple more eyes on this so we can get some more opinions and figure out what is the correct approach.

So lets just leave it for now, and we'll go from there if you're OK with that 👍